kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/driver/Driver.java

```diff
@@ -211,4 +211,26 @@ public void registerDriverListener(Map<String, Object> channelConfig, DriverList
      */
     public List<DriverRecord> write(List<DriverRecord> records) throws ConnectionException;
 
+    /**
+     * This method allows the driver to perform protocol specific optimizations in order to accelerate the execution of
+     * batches of read requests having the same channel configuration.
+     * The result of this optimization will be returned by the driver as a {@link PreparedRead} instance that can be
+     * used to perform the requests.
+     * In order to improve efficiency a driver should validate the channel configuration of the provided channels during
+     * this method call.
+     * It is also permitted to the implementation of the {@link PreparedRead#execute()} and
+     * {@link PreparedRead#getDriverRecords()} methods to return the same {@link DriverRecord} instances provided as an
+     * argument to this method.
+     * If the validation of the channel configuration fails for some channels, the driver must not throw an exception
+     * but it is required to return driver records with proper error flags set as a result of the
+     * {@link PreparedRead#execute()} call.
```

Comment:
Can we postpose this? looking at the javadocs there are also other places in which we could improve readability, maybe we can do this with a dedicated PR